pydy/viz/scene.py

```diff
@@ -48,71 +50,93 @@
+        states_symbols : sequence of functions of time, len(m), optional, default=None
+            An ordered sequence of the SymPy functions that represent the
+            states. The order must match the order of the
+            ``states_trajectories``.
+        states_trajectories : array_like, shape(n, m), optional, default=None
+            A two dimensional array with numerical values for each state at
+            each poitn in time during the animation.
+
+        Notes
+        =====
+        The user is allowed to supply either system or time, constants,
+        states_symbols, and states_trajectories. Providing a System allows for
+        interactively changing the simulation parameters via the Scene GUI
+        in the IPython notebook.
+
         """
 
         self.reference_frame = reference_frame
         self.origin = origin
         self.visualization_frames = list(visualization_frames)
-        self._system = None
-
-        try:
-            self.name = kwargs['name']
-        except KeyError:
-            self.name = 'unnamed'
 
-        try:
-            self.cameras = kwargs['cameras']
-        except KeyError:
-            camera_point = self.origin.locatenew('p_camera', 10 *
-                                                 self.reference_frame.z)
-            self.cameras = [PerspectiveCamera('DefaultCamera',
-                                              self.reference_frame,
-                                              camera_point)]
-
-        try:
-            self.lights = kwargs['lights']
-        except KeyError:
-            light_point = self.origin.locatenew('p_light', 10 *
-                                                self.reference_frame.z)
-            self.lights = [PointLight('DefaultLight', self.reference_frame,
-                                      light_point)]
+        vec = 10 * self.reference_frame.z
+        self._default_camera_point = self.origin.locatenew('p_camera', vec)
+        self._default_light_point = self.origin.locatenew('p_light', vec)
+
+        default_kwargs = {'name': 'unnamed',
+                          'cameras': [PerspectiveCamera('DefaultCamera',
+                                                        self.reference_frame,
+                                                        self._default_camera_point)],
+                          'lights': [PointLight('DefaultLight', self.reference_frame,
+                                                self._default_light_point)],
+                          'system': None,
+                          'time': None,
+                          'constants': None,
+                          'states_symbols': None,
+                          'states_trajectories': None,
+                          'frames_per_second': 30}
+        default_kwargs.update(kwargs)
+        for k, v in default_kwargs.items():
+            setattr(self, k, v)
+
+        self._static_url = os.path.join(os.getcwd(), "pydy-resources")
```

Comment:
I'm not sure that this is the best place to create the path to this dir. The user coudl theorectically change directory in between the calls to `__init__()` and `display()`.

I would make a class level attribute called `pydy_dir` or something. And then in the `display()` methods you should do this os.path.join.